OpenSim/Common/TimeSeriesTable.h

```diff
@@ -0,0 +1,158 @@
+/* -------------------------------------------------------------------------- *
+ *                            OpenSim:  TimeSeriesTable.h                     *
+ * -------------------------------------------------------------------------- *
+ * The OpenSim API is a toolkit for musculoskeletal modeling and simulation.  *
+ * See http://opensim.stanford.edu and the NOTICE file for more information.  *
+ * OpenSim is developed at Stanford University and supported by the US        *
+ * National Institutes of Health (U54 GM072970, R24 HD065690) and by DARPA    *
+ * through the Warrior Web program.                                           *
+ *                                                                            *
+ * Copyright (c) 2005-2015 Stanford University and the Authors                *
+ * Authors:                                                                   *
+ *                                                                            *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may    *
+ * not use this file except in compliance with the License. You may obtain a  *
+ * copy of the License at http://www.apache.org/licenses/LICENSE-2.0.         *
+ *                                                                            *
+ * Unless required by applicable law or agreed to in writing, software        *
+ * distributed under the License is distributed on an "AS IS" BASIS,          *
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.   *
+ * See the License for the specific language governing permissions and        *
+ * limitations under the License.                                             *
+ * -------------------------------------------------------------------------- */
+
+#ifndef OPENSIM_TIME_SERIES_DATA_TABLE_H_
+#define OPENSIM_TIME_SERIES_DATA_TABLE_H_
+
+/** \file
+This file defines the TimeSeriesTable_ class, which is used by OpenSim to 
+provide an in-memory container for data access and manipulation.              */
+
+#include "OpenSim/Common/DataTable.h"
+
+
+namespace OpenSim {
+
+class InvalidTable : public Exception {
+public:
+    using Exception::Exception;
+};
+
+class TimeColumnNotIncreasing : public InvalidTable {
+public:
+    TimeColumnNotIncreasing(const std::string& file, 
+                            size_t line,
+                            const std::string& func) :
+        InvalidTable(file, line, func) {
+        std::string msg = "Time column is not strictly increasing";
+
+        addMessage(msg);
```

Comment:
Consider Microsoft Word. Say you tried to print a document to PDF in Word but you didn't install PDF support in Word. I think we all agree it wouldn't make sense for Word to give a pop up error message that says: 

```
/home/francis/repos/word/Exporter.h:6316: In function "exportToPDF()": PDF support not available.
```

OpenSim is an application as much as it is a library. Shrinidhi's format is great for a library, but is unsuitable for an application. The issue right now is that we provide the *same* error messages in both the "application" and in the "library".

If we could format messages differently for the "application", that'd be great. One way would be for the application to catch the library's exceptions and print them in a different format.